### DISCUSSION TITLE
ESLint business model and funding tips

### DISCUSSION THREAD

**Discussion by malwinq** [eslint/eslint#16759] [General]
Hi! Together with @KatarzynaWojdalska we would love to ask a very non-technical question :blush:

## Background
Long story short - we're developing a tool called ec0lint. It's based on ESLint's engine (and stylelint's for CSS) and give frontend developers hints on what can be changed in order to create a more sustainable website. If you would like to know more, you can visit our [landing page](https://ec0lint.com/) or [GitHub](https://github.com/ec0lint).

## Question
We see ESLint collects many [donations](https://eslint.org/sponsors/) from both companies and ESLint community. It looks amazing and we really admire your work!

We were wondering if you could tell us how it works in details. How did you manage to achieve company donations and so nice crowd-funding? What did you begin with? Is it the main (only?) pillar of ESLint business model?

We're just starting our jurney, so we'll appreciate any tips about funding and maintaining open-source projects :)

**Comment by kecrily** [comment]
ESLint takes donations through [Open Collective](https://opencollective.com/eslint) and uses it to manage funds.

**Comment by malwinq** [comment]
Hi @kecrily 

We're aware eslint collects donations and manages them in Open Collective (we also have a profile there), however we're also aware that just having such a platform isn't enough - in fact. it gives nothing if it's not backed up with proper promotion and reaching out to the business. Each product needs to start a promotion campaign or/and offer something exceptionally interesting. 

We're curious if eslint authors could share their history with us: how they reached the current populatrity, if they encouraged companies to give them donations (how?), if they were looking for sponsors by contacting them directly (again - how?) or by using some specific platforms, e.g. LinkedIn. What was the initial point that led eslint to receive more and more donations? Did they offer some services, like consultation, to companies in exchange of larger donations?

Happy to hear any ideas :)

**Comment by nzakas** [comment]
@malwinq ESLint doesn't have any business model, per se, as we are just an open source project and not a business. Our first sponsors were from people approaching us directly, not us seeking them out. After that, it was just word of mouth. A lot of our sponsors come to us directly. We have a "Donate" button on every page of eslint.org to try to drive traffic but really, at this point, most of our donations come via reputation. People know, like, and appreciate ESLint, and want to support it. We've won several one-time grants from companies based on the recommendations of their employees, as well.

While I do sometimes reach out to companies directly if I notice them using ESLint in public, in reality, those have rarely turned into sponsorships. 

So, unfortunately, I don't think we have any real secrets to share other than to create a project that people like and believe in, and then mention that you're accepting donations. :)
